prov/usnic/src/usdf_fabric.c

```diff
@@ -116,13 +188,19 @@ usdf_validate_hints(struct fi_info *hints, struct usd_device_attrs *dap)
 		    fattrp->prov_version != USDF_PROV_VERSION) {
 			return -FI_ENODATA;
 		}
+
+		devn = get_devname(version, dap);
+
 		if (fattrp->name != NULL &&
-                    strcmp(fattrp->name, dap->uda_devname) != 0) {
-			return -FI_ENODATA;
+                    !check_devname(version, dap, NULL, CHECK_DEVNAME_GETINFO)) {
```

Comment:
You should be passing the `fattrp` variable in the third argument. Maybe you should make that take a string hint instead?